README.md

```diff
@@ -33,6 +33,285 @@ of this, and doing so can be easier said than done, especially if your algorithm
+
+There are various ways to handle these 'remainder' vectors. You could fall back to scalar code, or progressively fall back to narrower wide types, such as `Vec3x4`, or you can just consider whether the cost of calculating a few additional vectors that you won't use is worth adding complexity to your code.
+
+### Ray-Sphere Intersection
+
+Scalar code that operates on a single value at a time needs some restructuring to take advantage of SIMD and the 4-/8-wide data types.
+
+Below is an example of scalar ray-sphere instersection code using `Vec3` for points and vectors:
+
+```rust
+fn ray_sphere_intersect(
+    ray_o: uv::Vec3,
+    ray_d: uv::Vec3,
+    sphere_o: uv::Vec3,
+    sphere_r_sq: f32,
+) -> f32 {
+    let oc = ray_o - sphere_o;
+    let b = oc.dot(ray_d);
+    let c = oc.mag_sq() - sphere_r_sq;
+    let descrim = b * b - c;
+
+    if descrim > 0.0 {
+        let desc_sqrt = descrim.sqrt();
+
+        let t1 = -b - desc_sqrt;
+        if t1 > 0.0 {
+            t1
+        } else {
+            let t2 = -b + desc_sqrt;
+            if t2 > 0.0 {
+                t2
+            } else {
+                f32::MAX
+            }
+        }
+    } else {
+        f32::MAX
+    }
+}
+```
+
+This porting guide will not discuss the details of the algorithm, but will focus on how to convert the code to apply parallel SIMD operations on wide data types.
+
+The first thing to do is to convert the parameter types from scalar `Vec3` to wide `Vec3x8` and `f32x8`:
+
+```rust
+fn ray_sphere_intersect_x8(
+    ray_o: uv::Vec3x8,
+    ray_d: uv::Vec3x8,
+    sphere_o: uv::Vec3x8,
+    sphere_r_sq: uv::f32x8,
+) -> uv::f32x8 {
+```
+
+Each call to the function will process 8 ray-sphere intersections in parallel. The first four lines of the function remain the same:
+
+```rust
+    let oc = ray_o - sphere_o;
+    let b = oc.dot(ray_d);
+    let c = oc.mag_sq() - sphere_r_sq;
+    let descrim = b * b - c;
+```
+
+Despite this code being the same, the calculations for 8 rays and spheres will be carried out at the same time!
+
+The next line of the scalar code tests the value of `descrim` to see if it is greater than `0.0`. When operating on 8 values at a time, the code cannot branch along two separate paths because the value of `descrim` for each of the 8 values may cause branching to different sets of operations. To support this we would need to convert back to scalar code and then we lose all the performance benefits of parallel processing.
+
+So, how do we convert this? It turns out, the trick is to do the calculations for both branches for all 8 lanes, and then filter the results with masks that select the correct values from the possibilities at the end.
+
+To create the mask for 8 lanes of `descrim` values with `0.0`:
+
+```rust
+    let desc_pos = descrim.cmp_gt(uv::f32x8::splat(0.0));
+```
+
+In the true case, we do some more 8-lane parallel processing that looks like the scalar code:
```

Comment:
This line is a little confusing... perhaps something more like

"In the true case of the original scalar version, we have more arithmetic operations that end up looking the exact same when we do them on the vectorized version:"

Reply:
For some reason my brain didn’t want to pull forth “arithmetic operations” yesterday. That’s much clearer.

Reply:
I applied your suggestion here.